### PR TITLE
Mercury 13.0.0 plat7571

### PR DESCRIPTION
**Pull request by inbal-ben-david**

### CODE REVIEW

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 1]
```diff
@@ -769,9 +772,12 @@ public static function resizeEntryImage( entry $entry, $version , $width , $heig
 				// 3. the actual thumbnail doesnt exist on disk
 				($entry->getMediaType() == entry::ENTRY_MEDIA_TYPE_VIDEO && ($vid_sec != -1 || $vid_slices != -1))
 				||
+				($entry->getType() == entryType::PLAYLIST && ($vid_sec != -1 || $vid_slices != -1))
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: consolidate the checked terms.

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 2]
```diff
@@ -898,7 +909,74 @@ public static function resizeEntryImage( entry $entry, $version , $width , $heig
...
+	{
+		$mappedThumbEntryTypes = array(entryType::PLAYLIST);
+		if(in_array($entry->getType(), $mappedThumbEntryTypes))
+			return self::captureMappedThumbUsingPackager($entry, $capturedThumbPath, $calc_vid_sec, $flavorAssetId);
+
+		return self::captureLocalThumbUsingPackager($entry, $capturedThumbPath, $calc_vid_sec, $flavorAssetId);
+	}
+
+
+	public static function captureMappedThumbUsingPackager($entry, $capturedThumbPath, $calc_vid_sec, &$flavorAssetId)
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: change to private

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 3]
```diff
@@ -898,7 +909,74 @@ public static function resizeEntryImage( entry $entry, $version , $width , $heig
...
+	{
+		$packagerCaptureUrl = kConf::get('packager_mapped_thumb_capture_url', 'local', null);
+		if (!$packagerCaptureUrl)
+			return false;
+
+		if($entry->getType() == entryType::PLAYLIST)
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: no need to recheck.

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 4]
```diff
@@ -898,7 +909,74 @@ public static function resizeEntryImage( entry $entry, $version , $width , $heig
...
+			$success = self::curlThumbUrlWithOffset($flavorUrl, $calc_vid_sec, $packagerCaptureUrl, $capturedThumbPath);
+			if(!$success)
+				return false;
+
+			return true;
+		}
+	}
+
+
+	public static function curlThumbUrlWithOffset($url, $calc_vid_sec, $packagerCaptureUrl, $capturedThumbPath)
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: private

File: alpha/apps/kaltura/lib/myEntryUtils.class.php [hunk 5]
```diff
@@ -898,7 +909,74 @@ public static function resizeEntryImage( entry $entry, $version , $width , $heig
...
+	{
+		$packagerThumbCapture = str_replace(
+		array ( "{url}", "{offset}" ),
+		array ( $url , floor($calc_vid_sec*1000)  ) ,
+		$packagerCaptureUrl );
+
+		$tempThumbPath = $capturedThumbPath.'temp_1.jpg';
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/myEntryUtils.class.php]: Use const value for this name

File: alpha/apps/kaltura/lib/myPlaylistUtils.class.php [hunk 6]
```diff
@@ -1151,4 +1151,26 @@ protected static function getCaptionFilesForEntryIds($entryIds, $captionLanguage
...
+	public static function buildPlaylistThumbPath($entry, $flavorAsset)
+	{
+		$partnerId = $flavorAsset->getPartnerId();
+		$subpId = $entry->getSubpId();
+		$partnerPath = myPartnerUtils::getUrlForPartner($partnerId, $subpId);
+		$entryVersion = $entry->getVersion();
+
+		$url = "$partnerPath/serveFlavor/entryId/".$entry->getId();
+		$url .= ($entryVersion ? "/v/$entryVersion" : '');
+		$url .= "/flavorParamIds/" . $flavorAsset->getFlavorParamsId()."/name/a.mp4";
```
Comment (inbal-ben-david) [inline alpha/apps/kaltura/lib/myPlaylistUtils.class.php]: add const to myEntryUtils called VOD_GENERIC_FILE_SUFFIX

**Comment by MosheMaorKaltura** [comment]
Missing updates to release notes

**Pull request merged**